src/components/src/test/java/org/apache/jmeter/extractor/json/jmespath/TestJMESPathExtractor.java

```diff
@@ -0,0 +1,237 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package org.apache.jmeter.extractor.json.jmespath;
+
+import static org.junit.Assert.assertThat;
+
+import org.apache.jmeter.threads.JMeterContext;
+import org.apache.jmeter.threads.JMeterContextService;
+import org.apache.jmeter.threads.JMeterVariables;
+import org.hamcrest.CoreMatchers;
+import org.junit.Test;
+
+public class TestJMESPathExtractor {
+    private static final String DEFAULT_VALUE = "NONE"; // $NON-NLS-1$
+    private static final String REFERENCE_NAME = "varname"; // $NON-NLS-1$
+    private static final String REFERENCE_NAME_MATCH_NUMBER = "varname_matchNr"; // $NON-NLS-1$
+
+    private JMESPathExtractor setupProcessor(JMeterContext context, String matchNumbers) {
+        JMESPathExtractor processor = new JMESPathExtractor();
+        processor.setThreadContext(context);
+        processor.setRefName(REFERENCE_NAME);
+        processor.setMatchNumbers(matchNumbers);
+        processor.setDefaultValue(DEFAULT_VALUE);
+        processor.setScopeVariable("contentvar");
+        return processor;
+    }
+
+    @Test
+    public void testJMESPathExtractorZeroMatch() {
+        // test1
+        JMeterContext context = JMeterContextService.getContext();
+        JMESPathExtractor processor = setupProcessor(context, "-1");
+        JMeterVariables vars = new JMeterVariables();
+        context.setVariables(vars);
+        processor.setJmesPathExpression("a.b.c.f");
+        vars.put("contentvar", "{\"a\": {\"b\": {\"c\": {\"d\": \"value\"}}}}");
+        processor.process();
+        assertThat(vars.get(REFERENCE_NAME), CoreMatchers.is(DEFAULT_VALUE));
+        assertThat(vars.get(REFERENCE_NAME_MATCH_NUMBER), CoreMatchers.is("0"));
+    }
+
+    @Test
+    public void testJMESPathExtractorAllElementsOneMatch() {
```

Comment:
This seems to be a quite complex test, would it be better to split the tested components into simpler tests?